### ISSUE TITLE
Modifying a buffered property of a changeset invalidates all buffered properties 

### ISSUE THREAD

**Issue by mwpastore** [poteto/ember-changeset#437]
Got a weird one here. Not sure if it's intentional but I suspect not. Essentially, changing any buffered property of an ember-changeset causes the observers of all buffered properties to fire (i.e. the whole buffer is invalidated). If you have an underlying object with lots of properties and associated DOM, this can get real messy, real fast. I put together a basic repro that shows (clearly, I hope) the differences in behavior between a POJO and an ember-changeset. 

## Version
3.0.6

## Test Case
https://github.com/mwpastore/-ember-changeset-notify

![](https://media.giphy.com/media/ZZlmKrS9jt0JMAjyMk/giphy.gif)

## Steps to reproduce
Modify any buffered property of a changeset. 

## Expected Behavior
Only observers of the modified property should fire. 

## Actual Behavior
The observers of all properties fire.

**Comment by snewcomer** [comment]
@mwpastore Is this potentially due to the `#let` block emitting a new value.  For example, if I invert the problem - change `useChangeset = true`, toggle `Use changeset?`, both `Did Update` checkboxes are now checked.

Lmk if I am missing the goal of your repro though.

**Comment by mwpastore** [comment]
@snewcomer Yes, when you switch between a POJO and a changeset, it invalidates both properties. But that's not what I'm trying to show with my repro. The difference in behavior is most easily seen when both "Did Update?" checkboxes are un-checked—i.e. by manually un-checking them one at a time; or by clicking "(clear)" to do them both at the same time. 

Maybe a "script" would help. You can run through these steps, and feel free to play with the starting conditions. It does not matter whether "Use changeset?" is initially true or false. 

When "Use changeset?" is un-checked:

1. Click "(clear)" to un-check both "Did Update?" checkboxes
1. Toggle the "Value" checkbox under Foo
1. Note that the "Did Update?" checkbox under Foo is now checked, _but checkbox under Bar remains unchecked, because the value of Bar did not change_

When "Use changeset?" is checked:

1. Click "(clear)" to un-check both "Did Update?" checkboxes
1. Toggle the "Value" checkbox under Foo
1. Note that the "Did Update?" checkbox under Foo is now checked, _and the checkbox under Bar is also now checked, even though the value of Bar did not change_

**Comment by mwpastore** [comment]
@snewcomer This still appears to be an issue with Ember 3.18.1 and ember-changeset 3.3.8. Has my most recent comment helped explain the problem I'm seeing, or is it still unclear?

**Comment by snewcomer** [comment]
@mwpastore Can you try 3.4.0?  Thank you!!

**Comment by mwpastore** [comment]
@snewcomer No change. Do you have an inkling of what/where the problem is?

**Comment by snewcomer** [comment]
Not anymore.  We have overhauled the `get` access on a changeset so it should work completely with any level of nested properties.  I think what would be best to get to the core of the issue is one simple failing test in this repo or validated-changeset.

**Comment by snewcomer** [comment]
Hi @mwpastore!  Mind trying out the latest 3.6.0?

**Comment by mwpastore** [comment]
@snewcomer No change. I hope to have some time to work on that failing test next week. Cheers.

**Comment by snewcomer** [comment]
@mwpastore 👋 👋 Any updates?
